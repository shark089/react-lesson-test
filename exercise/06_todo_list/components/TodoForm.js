var TodoForm = React.createClass({
  onSubmit: function (e) {
    e.preventDefault();
    var text = this.input.value;
    this.input.value = '';
    this.props.addItem(text);
    console.log(this.input.value);
  },
  inputRef: function (input) {
    this.input = input;
  },
  render: function () {
    return (
      <form className="todo-form" onSubmit={this.onSubmit}>
        <input type="text" ref={this.inputRef}/>
      </form>
    );
  },
});
