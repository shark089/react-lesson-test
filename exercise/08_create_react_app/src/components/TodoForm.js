/* eslint-disable */
var TodoForm = React.createClass({
  onSubmit: function (e) {
    e.preventDefault();
    var text = this.input.value;
    this.input.value = '';
    this.props.addItem(text);
  },

  inputRef: function (input) {
    this.input = input;
  },

  render: function () {
    return (
      <form className="todo-form" onSubmit={this.onSubmit}>
        <input type="text" ref={this.inputRef} />
      </form>
    );
  },
});
