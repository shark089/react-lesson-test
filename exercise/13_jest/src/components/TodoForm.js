import React, { Component } from 'react';

class TodoForm extends Component {
  onSubmit = (e) => {
    e.preventDefault();
    const text = this.input.value;
    this.input.value = '';
    this.props.addItem(text);
  }

  inputRef = (input) => {
    this.input = input;
  }

  render() {
    return (
      <form className="todo-form" onSubmit={this.onSubmit}>
        <input type="text" ref={this.inputRef} />
      </form>
    );
  }
}

export default TodoForm;
