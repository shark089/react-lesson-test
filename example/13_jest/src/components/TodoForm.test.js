import React from 'react';
import { mount } from 'enzyme';

import TodoForm from './TodoForm';

const addItem = jest.fn();

const wrapper = mount(<TodoForm addItem={addItem} />);

it('should render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should prevent default', () => {
  const preventDefault = jest.fn();
  wrapper.simulate('submit', { preventDefault });
  expect(preventDefault).toHaveBeenCalled();
});

it('should call addItem', () => {
  wrapper.simulate('submit');
  expect(addItem).toHaveBeenCalledWith('');
});
