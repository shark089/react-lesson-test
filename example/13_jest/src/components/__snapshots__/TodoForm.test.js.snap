// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<TodoForm
  addItem={[Function]}
>
  <form
    className="todo-form"
    onSubmit={[Function]}
  >
    <input
      type="text"
    />
  </form>
</TodoForm>
`;
